contracts/core/libraries/QVHelper.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.19;
+
+import "solady/utils/FixedPointMathLib.sol";
+
+/// @title QV Helper Library
+/// @notice A helper library for Quadratic Voting
+/// @dev Handles the voting of recipients and calculates the payout amount for each recipient
+library QVHelper {
+    /// @notice Error thrown when the number of recipients and amounts are not equal on voting
+    error QVHelper_LengthMissmatch();
+
+    /// @notice Struct that holds the state of the voting
+    /// @param totalVotes The total amount of votes casted
+    /// @param totalVoiceCredits The total amount of voice credits casted
+    /// @param recipientVoiceCredits The voice credits casted for each recipient
+    /// @param recipientVotes The votes casted for each recipient
+    struct VotingState {
+        uint256 totalVotes;
+        uint256 totalVoiceCredits;
+        mapping(address => uint256) recipientVoiceCredits;
+        mapping(address => uint256) recipientVotes;
+    }
+
+    /// @notice Votes for recipients
+    /// @param _state The voting state
+    /// @param _recipients The recipients to vote
+    /// @param _voiceCredits The amounts of voice credits to for each recipient
+    /// @dev The number of recipients and voiceCredits should be equal and the same index should correspond to the same recipient and amount
+    function _voteWithVoiceCredits(
+        VotingState storage _state,
+        address[] memory _recipients,
+        uint256[] memory _voiceCredits
+    ) internal {
+        /// Check if the number of recipients and amounts are equal
+        if (_recipients.length != _voiceCredits.length) revert QVHelper_LengthMissmatch();
+
+        for (uint256 i = 0; i < _recipients.length; i++) {
+            /// Add the voice credits to the recipient
+            _state.recipientVoiceCredits[_recipients[i]] += _voiceCredits[i];
+            uint256 _votes = FixedPointMathLib.sqrt(_voiceCredits[i] * 1e18);
```

Comment:
Multiplying here by `1e18` makes `_voteWithVoiceCredits()` incompatible with `_vote()`. Is this intended?

I think we should either remove it or add it to `_vote()` as well, dividing `_voiceCredits` by `1e18`.